en/components/grids_templates/editing.md

```diff
@@ -288,6 +311,162 @@ These can be wired to user interactions, not necessarily related to the **@@igSe
 
 <div class="divider--half"></div>
 
+### Editing Events
+
+The grid exposes a wide array of events that provide greater control over the editing experience. These events are fired during the [**Row Editing**](row_editing.md) and **Cell Editing** life cycle - when starting, committing or canceling editing.
+
+The events can be broken down as follows:
+
+| Event | Description |
+|-------|-------------|
+| `onCellEditEnter` | Fires when a cell **enters edit mode** |
+| `onRowEditEnter` | If `[rowEditing]` is enabled, fires when a row **enters edit mode** |
+| `onCellEdit` | Fires when a cell's value is **committed** (e.g. by pressing `Enter`) |
+| `onCellEditCancel` | Fires when a cell exits edit mode **without committing** its value (e.g. by pressing `Escape`) |
+| `onRowEdit` | Fires when a row in edit mode's value is **committed** (e.g. by clicking the `Done` button on the Row Editing Overlay) |
+| `onRowEditCancel` | Fires when a row exits edit mode **without committing** its values (e.g. by clicking the `Cancel` button on the Row Editing Overlay) |
+
+All of the above events can be cancelled. For example, if `onCellEditEnter` is cancelled, the cell will never enter edit mode.
+
+#### Editing Events - Configuration
+Using the grid's editing events we can alter how the user interacts with the grid.
+In the below example, we'll validate a cell based on the data entered in it by binding to the [`onCellEdit`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#oncelledit) event. If the new value of the cell does not meet our predefined criteria, we'll display a custom error message (using [`IgxToast`](../toast.md). We'll also cancel the editing event (by setting `event.cancel = true`), so the user has to specify a valid value before continuing with the editing.
+
+The first thing we need to is bind to the grid's event:
+
+```html
+<@@igSelector (onCellEdit)="handleCellEdit($event)"
+...>
+...
+</@@igSelector>
+```
+@@if (igxName === 'IgxTreeGrid') {
```

Comment:
_That's Hierarchical Grid config below, no?_

Reply:
Also see https://github.com/IgniteUI/igniteui-angular/issues/5611 - this won't be the case for too long, so I think avoiding this section might actually be better. Plus the `columnList` lookup in the snippet after will only work on the root grid anyways.

Reply:
Correct, it's for the HGrid, thanks for pointing it out!